src/main/java/core/basesyntax/Fruits.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class Fruits {
+    private Map<String, Integer> fruitQuantities;
+
+    public Fruits() {
+        fruitQuantities = new HashMap<>();
+    }
+
+    public int balance(String fruitType,int init) {

```

Comment:
The method name `balance` is not very descriptive of its functionality. Consider renaming it to `initializeFruitQuantity` to better reflect its purpose.